lib/internal/util/types.js

```diff
@@ -54,8 +57,16 @@ function isBigUint64Array(value) {
   return TypedArrayPrototypeGetSymbolToStringTag(value) === 'BigUint64Array';
 }
 
+function isArrayBufferDetached(value) {
+  if (ArrayBufferPrototypeGetByteLength(value) === 0) {
```

Comment:
Should we do this before checking if `value` is an instance of `ArrayBuffer`?

Reply:
If the user passes something that's not an `ArrayBuffer`, instead of returning `false` the function would throw a `TypeError`, which is surprising. We should at least document this behavior if we want to keep it (another option is to not expose it publicly and keep it internal, or expose a slower version that first checks if the parameter is an `ArrayBuffer` before calling the internal util).